src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -114,6 +113,91 @@
+     * to the original nodes) and to reanchor any relationships from an old root node (which will not be cloned) to a different existing root node.
+     *
+     */
+    @Procedure(mode = Mode.WRITE)
+    @Deprecated
+    @Description("apoc.refactor.cloneSubgraph([node1,node2,...], [rel1,rel2,...], [reanchorNodePairs = [] ], [skipProperties = [] ]) YIELD input, output, error | " +
+            "clone nodes with their labels and properties (optionally skipping specific properties), and clone the given relationships (will exist between cloned nodes only or be reanchored according to reanchor node pairings)")
+    public Stream<NodeRefactorResult> cloneSubgraph(@Name("nodes") List<Node> nodes,
+                                                    @Name("rels") List<Relationship> rels,
+                                                    @Name(value = "reanchors", defaultValue = "[]") List<List<Node>> reanchors,
+                                                    @Name(value = "skipProperties", defaultValue = "[]") List<String> skipProperties) {
+        Map<Node, Node> copyMap = new LinkedHashMap<>(nodes.size());
+        Map<Node, Node> reanchorMap = reanchors.isEmpty() ? Collections.emptyMap() : new LinkedHashMap<>(reanchors.size());
+        List<NodeRefactorResult> resultStream = new ArrayList<>();
+        if (nodes == null) return Stream.empty();
+
+        // populate reanchor map
+        for (List<Node> pairing : reanchors) {
+            if (pairing == null) continue;
+
+            if (pairing.size() != 2) {
+                throw new IllegalArgumentException("\'reanchors\' must be a list of node pairs");
+            }
+
+            Node from = pairing.get(0);
+            Node to = pairing.get(1);
+
+            if (from == null || to == null) {
+                throw new IllegalArgumentException("\'reanchors\' must be a list of node pairs");
+            }
+
+            reanchorMap.put(from, to);
+        }
+
+        // clone nodes and populate copy map
+        for (Node node : nodes) {
+            if (node == null || reanchorMap.containsKey(node)) continue;
+            // nodes being reanchored from will NOT be cloned
+
+            NodeRefactorResult result = new NodeRefactorResult(node.getId());
+            try {
+                Node copy = copyLabels(node, db.createNode());
+
+                Map<String, Object> properties = node.getAllProperties();
+                if (skipProperties != null && !skipProperties.isEmpty()) {
+                    for (String skip : skipProperties) properties.remove(skip);
+                }
+                copy = copyProperties(properties, copy);
+
+                resultStream.add(result.withOther(copy));
+                copyMap.put(node, copy);
+            } catch (Exception e) {
+                resultStream.add(result.withError(e));
+            }
+        }
+
+        // clone relationships, will be between cloned nodes and/or reanchors
```

Comment:
how easy is it actually to provide the relationships within the structure?

I originally thought we would clone all rels _within_ the structure automatically?

Reply:
You could use apoc.algo.cover() for that. I wanted to support cases where we could be selective of which relationships get cloned. Other methods to get these relationships, such as via the path expander procs, may be more efficient.

Maybe we could say that if no relationships are provided that we automatically get all the relationships between the cloned nodes?

Reply:
yep or a config option.

Reply:
Okay, made `rels` optional, if missing or an empty list we'll internally run `apoc.algo.cover()` to get all the rels between the nodes and use that. Test cases updated too. Will update docs accordingly.